### PR TITLE
Hotfix/stats cache

### PR DESCRIPTION
**Pull request by ixm-759**
Add collection to save statistics in database.
In case there is no cached information, it is obtained from the database.
Save / update / delete statistics data from the database.

### CODE REVIEW

File: src/stats/stats.controller.ts [hunk 1]
```diff
@@ -29,6 +29,6 @@ export class StatsController {
   async getStatsForWallet(@Param('wallet') wallet: string): Promise<any> {
     this.logger.debug('Called GET /stats/:wallet');
     return [];
-    // return this.statsService.getStatsForWallet(wallet);
+    //return this.statsService.getStatsForWallet(wallet);
```
Comment (ape-swap) [inline src/stats/stats.controller.ts:32]: Feel free to add this back.

File: src/stats/stats.service.ts [hunk 2]
```diff
@@ -35,10 +41,46 @@ export class StatsService {
...
+    const stat: any = await this.findOne();
+    if (!stat) return null;
+
+    const lastCreatedAt = new Date(stat.createdAt).getTime();
+    const diff = now - lastCreatedAt;
+    const time = 300000; //3 minutos
+
+    if (diff > time) return null;
+
+    return true;
```
Comment (ape-swap) [inline src/stats/stats.service.ts:79]: If this function would to return a StatsDocument or null, when it returns the document you could go ahead and use it to save an additional quey.

File: src/stats/stats.service.ts [hunk 3]
```diff
@@ -92,13 +134,24 @@ export class StatsService {
...
-  async calculateStats() {
+  async verifyCalculateStats() {
     const cachedValue = await this.cacheManager.get('calculateStats');
     if (cachedValue) {
       this.logger.log('Hit calculateStats() cache');
       return cachedValue as GeneralStats;
     }
+    const infoStats = await this.verifyStats();
+    this.logger.log(`infostats is ${infoStats}`);
+    if (infoStats == null) {
```
Comment (ape-swap) [inline src/stats/stats.service.ts:145]: If you make the change I previously comment you can save yourself an additional DB query and simplify the code a little bit. 

```javascript
const infoStats = await this.verifyStats();
this.logger.log(`infostats is ${infoStats}`);
if (infoStats) return infoStats;

await this.updateCreatedAtStats();
this.calculateStats();
const generalStats: any = await this.findOne();
return generalStats;
```

**Review by ape-swap — CHANGES_REQUESTED** [review]
Excellent work! Just some minor observations!

File: src/stats/stats.controller.ts (hunk 1 above)
Comment (ixm-759) [reply src/stats/stats.controller.ts]: Done

File: src/stats/stats.service.ts (hunk 2 above)
Comment (ixm-759) [reply src/stats/stats.service.ts]: Done

File: src/stats/stats.service.ts (hunk 3 above)
Comment (ixm-759) [reply src/stats/stats.service.ts]: Done

Comment (ixm-759) [reply src/stats/stats.service.ts]: Sorry, I did not see this comment well, I think I did not adapt it correctly. I check it again

Comment (ixm-759) [reply src/stats/stats.service.ts]: I did it in another way, putting that direct logic in the other method.

**Review by ape-swap — APPROVED** [review]
Excellent work!

**Pull request merged**